### PR TITLE
Improve clipping in pyramid creation

### PR DESCRIPTION
**Pull request by jluethi**
Currently, the percentile clipping has a hard-coded lower limit for clipping. This is reasonable, but there is a better choice for the actual limit. Because the 16 bit images are rescaled to 8bit for the viewer, the lowest value that should be rescaled is such that there are 256 levels in the final image. In that way, if there is low signal, then the original intensities are maintained in the viewer and the user can see finer gradients. 
Currently, this threshold is arbitrarily at 700. This patch sets it to 256 levels from background.
Also, the behavior for clipping with user-defined upper clipping thresholds is improved (background is set to 0.001 percentile instead of 0).
I tested this code on my develop instance. It improve the rescaling in dim images and did not make any issues.

### CODE REVIEW

File: tmlib/workflow/illuminati/api.py [hunk 1]
```diff
@@ -148,16 +148,17 @@ def create_run_batches(self, args):
                             if layer.channel.bit_depth == 8:
                                 if clip_max < 255:
                                     clip_max = 255
+                                    
+                            clip_min = stats.get_closest_percentile(0.001)
```
Comment (riccardomurri) [inline tmlib/workflow/illuminati/api.py]: This line looks badly indented w.r.t. to following `else:` at line 153.

File: tmlib/workflow/illuminati/api.py [hunk 2]
```diff
@@ -148,16 +148,17 @@ def create_run_batches(self, args):
...
                             else:
-                                if clip_max < 700:
-                                    clip_max = 700
+                                if clip_max < clip_min + 255:
```
Comment (riccardomurri) [inline tmlib/workflow/illuminati/api.py]: This code belongs to the `else:` branch of `if layer.channel.bit_depth == 8`, so we know that `layer.channel.bit_depth > 8`, right?  So why constrain `clip_max - clip_min == 255`?  Shouldn't we be able to use more than 256 levels of gray here?

File: tmlib/workflow/illuminati/api.py [hunk 3]
```diff
@@ -148,16 +148,17 @@ def create_run_batches(self, args):
...
+                                    clip_max = clip_min + 255
                             logger.info('clip value: %d', clip_max)
-                            clip_min = stats.get_closest_percentile(0.001)
                         else:
                             logger.info('use provided clip value')
                             clip_max = args.clip_value
                             logger.info('clip value: %d', clip_max)
-                            clip_min = 0
+                            clip_min = stats.get_closest_percentile(0.001)
```
Comment (riccardomurri) [inline tmlib/workflow/illuminati/api.py]: I read this as kind of "get closest as possible to zero while still non-zero".  Wouldn't (something like) `clip_min = stats.min()` be better?

Comment (jluethi) [reply tmlib/workflow/illuminati/api.py]: The 0.001 percentile is a save against dead pixels. There may be very rare pixel with value 0, but we wouldn't care about those normally. 0.001 percentile is typically even too conservative, but it's save.

**Comment by riccardomurri** [comment]
Great! Thank you very much for contributing this!

**Pull request merged**